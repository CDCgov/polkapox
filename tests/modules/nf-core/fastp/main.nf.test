nextflow_process {

    name "FASTP Module Tests"
    script "../../../../modules/nf-core/fastp/main.nf"
    process "FASTP"
    tag "fastp"

    test("paired_end_basic") {
        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'test_sub', single_end: false ],
                    [
                        file("${projectDir}/tests/nf-test-data/test_sub_1.fq.gz", checkIfExists: true),
                        file("${projectDir}/tests/nf-test-data/test_sub_2.fq.gz", checkIfExists: true)
                    ]
                ])
                input[1] = false
                input[2] = false
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

}
