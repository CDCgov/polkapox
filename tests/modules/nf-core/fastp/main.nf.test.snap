{
    "paired_end_stub": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    
                ],
                "6": [
                    
                ],
                "html": [
                    
                ],
                "json": [
                    
                ],
                "log": [
                    
                ],
                "reads": [
                    
                ],
                "reads_fail": [
                    
                ],
                "reads_merged": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-06-24T14:40:32.052631015"
    },
    "paired_end_basic": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    
                ],
                "6": [
                    
                ],
                "html": [
                    
                ],
                "json": [
                    
                ],
                "log": [
                    
                ],
                "reads": [
                    
                ],
                "reads_fail": [
                    
                ],
                "reads_merged": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-06-24T14:45:37.499556435"
    },
    "paired_end_with_merge": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    
                ],
                "6": [
                    
                ],
                "html": [
                    
                ],
                "json": [
                    
                ],
                "log": [
                    
                ],
                "reads": [
                    
                ],
                "reads_fail": [
                    
                ],
                "reads_merged": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-06-24T14:35:22.045464211"
    },
    "paired_end_real_sample": {
        "content": [
            [
                
            ],
            [
                
            ],
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-06-24T14:40:26.722831682"
    },
    "paired_end_with_fail_trim": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    
                ],
                "6": [
                    
                ],
                "html": [
                    
                ],
                "json": [
                    
                ],
                "log": [
                    
                ],
                "reads": [
                    
                ],
                "reads_fail": [
                    
                ],
                "reads_merged": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-06-24T14:45:43.159121847"
    }
}